algorithms/src/Kokkos_Sort.hpp

```diff
@@ -456,18 +456,37 @@ class BinSort {
   void operator()(const bin_sort_bins_tag& /*tag*/, const int i) const {
     auto bin_size = bin_count_const(i);
     if (bin_size <= 1) return;
+    constexpr bool use_std_sort =
+#ifdef KOKKOS_ENABLE_SERIAL
+        std::is_same_v<exec_space, Kokkos::Serial> ||
+#endif
+#ifdef KOKKOS_ENABLE_OPENMP
+        std::is_same_v<exec_space, Kokkos::OpenMP> ||
+#endif
+        false;
```

Comment:
Is there a particular reason to only check for these two? Otherwise, I would suggest:
```suggestion
    constexpr bool use_std_sort = Kokkos::SpaceAccessibility<exec_space, Kokkos::HostSpace>::accessible;
```

Reply:
The user still decides where to execute the sort. AFAICT, this pull request doesn't change that at all.

Reply:
I'm fine with whatever you guys think is the best option. I have no experience with the backends other than Serial and OpenMP, and those are the ones I really care about. Feel free to push into the branch what you would like it to be.

Reply:
I think maybe something to the affect of `is_same_v<exec_space::memory_space, HostSpace>` is the thing we may need to use here?